src/OpenTracing/Propagation/HttpHeadersCarrier.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Collections.Generic;
+using System.Net.Http.Headers;
+
+namespace OpenTracing.Propagation
+{
+    /// <summary>
+    /// A <see cref="ITextMapCarrier"/> which allows <see cref="HttpHeaders"/> implementations to be used as carrier objects.
+    /// </summary>
+    /// <remarks>
+    /// <see cref="HttpHeaders"/> is a multi-value dictionary. Since most other platforms represent http headers as regular
+    /// dictionaries, this carrier represents it as a regular dictionary to tracer implementations.</remarks>
+    public class HttpHeadersCarrier : ITextMapCarrier
+    {
+        // TODO should this class be internal/private? It's name is misleading because this only works with the .NET framework class "HttpHeaders"
```

Comment:
ah, I just saw this comment; this should only be part of the core OT API if.f. the .NET HttpHeaders is nearly ubiquitous.